src/elements/checkbox/checkbox.js

```diff
@@ -0,0 +1,59 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('semantic.ui.elements.checkbox', [])
+
+    .directive('smCheckbox', smCheckbox);
+
+  function smCheckbox() {
+    return {
+      restrict: 'E',
+      require: '?ngModel',
+      transclude: true,
+      scope: {
+        ngDisabled: '='
+      },
+      template: '<div class="ui checkbox" ng-class="{ checked: isChecked }">' +
```

Comment:
hey, guys. I think @stevesherrie's attempt looks pretty decent, we can improve some stuff later on. I may only suggest to add "replace: true" option in the directive definition object.